### PR TITLE
Error for empty files

### PR DESCRIPTION
**Pull request by david-nishi**
fix for issue #454

Notable changes: 
- added new error code 98: EMPTY_FILE
- added code to push error in testFile() (already called on config, bval, bvec, JSON, and tsv files)
- added misc group to files in fullTest to collect everything not run through testFile()
- added validateMisc(), which runs misc files through testFile()

### CODE REVIEW

File: utils/files/testFile.js [hunk 1]
```diff
@@ -5,34 +5,47 @@ const Issue = require('../../utils/issues').Issue
...
+          if (!accessErr) {
+            process.nextTick(function() {
+              if (stats.size === 0) {
+                resolve({
+                  issue: new Issue({
+                    code: 97,
+                    file: file,
+                    reason: `Empty files (${
+                      file.path
+                    }) not allowed; may corrupt archives.`,
```
Comment (nellh) [inline utils/files/testFile.js]: The "may corrupt archives" part was due to a SciTran bug, not relevant to the validator.

**Review by nellh — CHANGES_REQUESTED** [review]
This should use a smaller example dataset or the test data should go in bids-examples, this adds a lot of files to the repo for one test suite.

**Comment by david-nishi** [comment]
working on merge conflict in testFile.js

File: utils/files/testFile.js (hunk 1 above)
Comment (david-nishi) [reply utils/files/testFile.js]: removed "may corrupt archives" and minimized test file

**Review by nellh — APPROVED** [review]

**Pull request merged**